content/en/platform/corda/5.0/developing-applications/api/api-crypto-extensions.md

```diff
@@ -0,0 +1,230 @@
+---
+date: '2023-02-10'
+version: 'Corda 5.0 Beta 4'
+title: "net.corda.v5.crypto.extensions"
+menu:
+  corda5:
+    identifier: corda5-api-crypto-extensions
+    parent: corda5-api
+    weight: 5050
+section_menu: corda5
+---
+# net.corda.v5.crypto.extensions
+The `corda-crypto-extensions` module of the `Corda Crypto API` defines low-level crypto capabilities that can be used to extend functionality of the Corda Crypto Library by implementing them in a CPK. 
+
+# Extending Supported Digest Algorithms
+
+The Corda Crypto Library implements a wide variety of digest algorithms out of the box. However, you can use a digest algorithm that is not supported by the library by simply implementing some interfaces and adding the code into the CPK with the CorDapp code. Corda picks up any custom algorithms at runtime. We recommend adding custom digest code in a separate Java module.
+
+{{< note >}}
+* Digest algorithms must be cryptographically strong. For example, MD5 is not a strong algorithm and so is not supported by the library.
+* Custom algorithms cannot be used as an implicit part of the digital signing. For example, you cannot specify a signature specification such as 'SHA-256-TRIPLEwithRSA'. You must calculate the digest first and then sign/verify the produced hash using built-in signature specs.
```

Comment:
```suggestion
* Custom algorithms cannot be used as an implicit part of the digital signing. For example, you cannot specify a signature specification such as 'SHA-256-TRIPLEwithRSA'. You must calculate the digest first and then sign/verify the produced hash using built-in signature specifications.
```

Reply:
@kyriathar are you ok with this change?
Or should it be (or similar)
```suggestion
* Custom algorithms cannot be used as an implicit part of the digital signing. For example, you cannot specify a signature specification such as 'SHA-256-TRIPLEwithRSA'. You must calculate the digest first and then sign/verify the produced hash using a built-in `SignatureSpec`.
```

Reply:
Actually very good spot. 

No, we currently don't offer to users signing using custom digest algorithms. Nor they should be implementing `SignatureSpec`. So actually there is no way for them to define a custom signature specification (like the mentioned one _SHA-256-TRIPLEwithRSA_). This text must be obsolete from the time where they could, back then, to declare `SignatureSpec`s (now they need to go through the `SignatureSpecService` to acquire one). So actually this entire paragraph should be removed. 

Custom digest algorithms defined by users (through `crypto-exceptions` module) will only be available to be used for hashing only, and not signing.